rasa/nlu/utils/data_utils.py

```diff
@@ -0,0 +1,27 @@
+from typing import List, Optional, Text
+from rasa.shared.nlu.training_data.message import Message
+
+
+def remove_unfeaturized_messages(
+    messages: List[Message], attribute: Text, featurizers: Optional[List[Text]]
+) -> List[Message]:
+    """Removes messages that don't have required features.
+
+    Some NLU components require messages to have specific features to
+    make training and prediction possible. If we don't filter out messages
+    which don't have needed features, it will lead to errors.
+
+    Args:
+        messages: List of messages we want to filter out.
+        attribute: Message attribute.
+        featurizers: Names of featurizers to consider.
+
+    Returns:
+        List of messages where unfeaturized messages are removed.
+    """
+    filtered_messages = [
+        message
+        for message in messages
+        if message.features_present(attribute=attribute, featurizers=featurizers)
+    ]
+    return filtered_messages
```

Comment:
I appreciate that you tried to remove complexity and added the docstring with the explanation + the unit test 🚀 , but I think in this case the extra layer of abstraction is not really needed: 
- During prediction, we can just call `message.features_present(attribute=attribute, featurizers=featurizers)` directly. 
- During training, the extra lines don't hurt much and it is more transparent/explicit (i.e. easier to notice that we only check whether any of the requested featurizers produced something and not all of them - which is nit-picking but might become relevant for some cases later)
- Moreover, we already have tests for `features_present` and so what this really only tests is that the listcomp is implemented ok (which we get for free via the new `test_process_unfeatrized` ...)
Does that make sense?

Reply:
Yeap, definitely makes sense. Now that I think about it I'm not even sure why I decided to go for this tbh :smile: